embedded/Sentinel Embedded/src/test_mqtt.cpp

```diff
@@ -0,0 +1,140 @@
+/**************************************************************
+ *
+ * Important libraries used:
+ * PubSubClient
+ * https://github.com/knolleary/pubsubclient
+ *
+ * Documentation
+ * https://pubsubclient.knolleary.net/api
+ *
+ **************************************************************
+ * This test connects to HiveMQ's showcase broker.
+ *
+ * We can test sending and receiving messages from the HiveMQ webclient
+ * available at http://www.hivemq.com/demos/websocket-client/.
+ *
+ * Subscribe to the topic GsmClientTest/ledStatus
+ * Publish "toggle" to the topic GsmClientTest/led and the LED on your board
+ * should toggle and you should see a new message published to
+ * GsmClientTest/ledStatus with the newest LED status.
+ *
+ **************************************************************/
+#include <Arduino.h>
+#include <Modem.h>
+
+// See all AT commands, if wanted
+// #define DUMP_AT_COMMANDS
+
+// Range to attempt to autobaud
+// NOTE:  DO NOT AUTOBAUD in production code.  Once you've established
+// communication, set a fixed baud rate using modem.setBaud(#).
+#define GSM_AUTOBAUD_MIN 9600
+#define GSM_AUTOBAUD_MAX 115200
+
+// Add a reception delay, if needed.
+// This may be needed for a fast processor at a slow baud rate.
+// #define TINY_GSM_YIELD() { delay(2); }
+
+// MQTT details
+const char *broker = "broker.hivemq.com";
+const char *topicLED = "GsmClientTest/led";
+const char *topicInit = "GsmClientTest/init";
+const char *topicLEDStatus = "GsmClientTest/ledStatus";
+
+#include <TinyGsmClient.h>
+#include <PubSubClient.h>
+
+#ifdef DUMP_AT_COMMANDS
+#include <StreamDebugger.h>
+StreamDebugger debugger(SerialAT, SerialMon);
+TinyGsm modem(debugger);
+#else
+TinyGsm modem(SerialAT);
+#endif
+
+TinyGsmClient client(modem);
+PubSubClient mqtt(client);
+
+#define LED_PIN 13
+int ledStatus = LOW;
+uint32_t lastReconnectAttempt = 0;
+float lat, lon = 0;
```

Comment:
Don't need these, was here for some testing, but not bothered to remove this in this diff, will do it next time.

Testing concluded that GPS, GPRS (cellular GPS) is hopeless and we will need to buy a separate GPS module! :(